### PR TITLE
Update solidity contents to vyper

### PR DESCRIPTION
**Pull request by iamonuwa**
Refs #45

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
Thanks @iamonuwa I'm going to make a few changes to these files, mostly to replace the online IDE with Remix, as it now supports Vyper, and fix some grammar issues that were in the original posts, so these weren't things part of the original brief, and I'll handle them and let you know when I'm done.

That said, the vyper alternative of https://kauri.io/article/f95f956261494090be1aaa8227464773/v8/truffle:-testing-your-smart-contract seems to be missing, was that not possible to convert?

File: solidity-vyper-updates/Vyper Online Compiler - Your first smart contract.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+We add the public decorator to the function so that it can be called from the contract. In order to send ETH to our contract we need to add the payable keyword to our function. Without this payable keyword the contract will reject all attempts to send ETH to it via this function. Read more about [decorators](https://vyper.readthedocs.io/en/latest/structure-of-a-contract.html#functions).
+
+```
+def issueBounty(_data: bytes32, _deadline: timestamp):
+    assert msg.value > 0
+    assert _deadline > block.timestamp
+
+    bIndex: int128 = self.nextBountyIndex
```
Comment (ChrisChinchilla) [inline solidity-vyper-updates/Vyper Online Compiler - Your first smart contract.md:121]: @iamonuwa There were a few code errors, most I managed to fix, but I am stuck on this one. At this line I get:

```
line 34: Persistent variable undeclared: nextBountyIndex bIndex: int128 = self.nextBountyIndex
```

**Comment by iamonuwa** [comment]
Am working on the testing at the moment

File: solidity-vyper-updates/Vyper Online Compiler - Your first smart contract.md (hunk 1 above)
Comment (iamonuwa) [reply solidity-vyper-updates/Vyper Online Compiler - Your first smart contract.md:121]: I'll take a look

Comment (ChrisChinchilla) [reply solidity-vyper-updates/Vyper Online Compiler - Your first smart contract.md:121]: @iamonuwa Thanks!

**Comment by ChrisChinchilla** [comment]
@iamonuwa Sorry, still getting errors in the remix / first smart contract post.

```
line 31:21 Persistent variable undeclared: nextBountyIndex
     30
---> 31     bIndex: int128 = self.nextBountyIndex
-----------------------------^
     32
```

**Comment by iamonuwa** [comment]
> @iamonuwa Sorry, still getting errors in the remix / first smart contract post.

Please can you give me the steps to reproduce? I'll try and reproduce it and fix it. ATM, I can't reproduce the error locally

**Comment by ChrisChinchilla** [comment]
@iamonuwa I take the code from bounties.vy and try to compile it, in Remix, Vyper compiler, on the command line, the VSC extension, all show the same error.

**Comment by ChrisChinchilla** [comment]
@iamonuwa Maybe something is missing in the examples, if you can add the full code somewhere we can check :)

**Comment by iamonuwa** [comment]
> @iamonuwa Maybe something is missing in the examples, if you can add the full code somewhere we can check :)

I'll build a dapp for this, this weekend

**Comment by ChrisChinchilla** [comment]
@iamonuwa OK, do you not have the full contract that you wrote while building the tutorials?

**Comment by iamonuwa** [comment]
> @iamonuwa OK, do you not have the full contract that you wrote while building the tutorials?

I do, I'll write a dapp to interact with it.

**Comment by ChrisChinchilla** [comment]
Any updates @iamonuwa Once we have working code we can get this wrapped up :)

**Comment by iamonuwa** [comment]
> Any updates @iamonuwa Once we have working code we can get this wrapped up :)

It will be available today. Meanwhile, this is the repo for the contract https://github.com/iamonuwa/Bounties

**Comment by ChrisChinchilla** [comment]
@iamonuwa This was enough for me to figure out the problem, your tutorials miss adding these two lines - https://github.com/iamonuwa/Bounties/blob/73ec0fcc54955e50cb0d727e1d84d7af9caefe10/contracts/Bounties.vy#L25

**Comment by iamonuwa** [comment]
My bad, need to update it on the tutorial now. Apologises for that

**Comment by iamonuwa** [comment]
@ChrisChinchilla any update?

**Comment by ChrisChinchilla** [comment]
@iamonuwa Reviewing now. PS. When you force push you sometimes overwrite changes I made, so be careful with that…

**Comment by ChrisChinchilla** [comment]
OK, @iamonuwa I made quite a lot small changes, please don't force push and overwrite them :)

The last things I need from you is you'll find a few TODOs to update the images and links to your code, if you could add those and we're good to go. Once we've published we can update the links to each article.

**Comment by iamonuwa** [comment]
@ChrisChinchilla please take a look at the update. Thanks

**Comment by ChrisChinchilla** [comment]
OK @iamonuwa Let's get this merged, I will have to make some changes after you have published, so watch out for notifications on this via kauri.

**Comment by iamonuwa** [comment]
Do you need me to publish it via kauri?

**Comment by ChrisChinchilla** [comment]
@iamonuwa Yes please! It needs to be with your username :)

**Comment by ChrisChinchilla** [comment]
@iamonuwa Any updates?

**Comment by iamonuwa** [comment]
Awesome,  doing that now

**Comment by iamonuwa** [comment]
Done.

1. https://kauri.io/article/195c5784663e4963b16d914900ba5cf5/v5/understanding-smart-contract-compilation-and-deployment
2. https://kauri.io/article/15b95281c17a4954a5278865bbc0f515/v4/remix-ide-your-first-vyper-smart-contract
3. https://kauri.io/article/ebc4a29cc1c044fdba99631796ffbd93/v2/truffle:-testing-your-smart-contract
4. https://kauri.io/article/1ac9d10358b94945b06b9c893cd5bfcf/v2/truffle:-smart-contract-compilation-and-deployment
5. https://kauri.io/article/ca43e064d5054f829f733970909de1ad/v1/tools-for-working-with-vyper

**Pull request merged**